### PR TITLE
Review3

### PR DESCRIPTION
**Pull request by nattawutk123**

### CODE REVIEW

File: lib/features/sign_in/presentation/sign_in_page.dart [hunk 1]
```diff
@@ -80,12 +113,26 @@ class _SignInPageState extends State<SignInPage> {
...
+                    bool success = await signIn(
+                        _emailController.text, _passwordController.text);
+                    if (success) {
+                      // ignore: use_build_context_synchronously
+                      Navigator.push(
+                        context,
+                        MaterialPageRoute(
+                            builder: (context) =>
+                                const HomePage(initialIndex: 0)),
+                      );
```
Comment (GoKubo1997) [inline lib/features/sign_in/presentation/sign_in_page.dart]: If you want to use the context in an async function, you can do like...

```dart
if (context.mounted) {
  Navigator.push(...)
}
```

File: lib/features/splash/presentation/splash_page.dart [hunk 2]
```diff
@@ -25,10 +27,21 @@ class _SplashPageState extends State<SplashPage>
...
+        Navigator.of(context).pushReplacement(
+          MaterialPageRoute(
+              builder: (context) => const HomePage(initialIndex: 0)),
+        );
+      } else {
+        // ignore: use_build_context_synchronously
+        Navigator.of(context).pushReplacement(
+          MaterialPageRoute(builder: (context) => const SignInPage()),
+        );
+      }
```
Comment (GoKubo1997) [inline lib/features/splash/presentation/splash_page.dart]: Or, you can improve readability just a little bit like...

```dart
final hasSignedIn = token != null && token.isNotEmpty;
final nextPage = hasSignedIn? const HomePage(initialIndex: 0) : const SignInPage();

  Navigator.of(context).pushReplacement(
    MaterialPageRoute(builder: (context) => nextPage),
  );
```

File: lib/widgets/random_book.dart [hunk 3]
```diff
@@ -76,6 +76,7 @@ class _RandomBookWidgetState extends State<RandomBookWidget> {
                       book!.category,
                       style: const TextStyle(
                         color: Colors.white,
+                        //fontSize: 25,
```
Comment (GoKubo1997) [inline lib/widgets/random_book.dart]: Please delete it.

File: lib/features/sign_in/presentation/sign_in_page.dart [hunk 4]
```diff
@@ -31,6 +35,35 @@ class _SignInPageState extends State<SignInPage> {
...
+      },
+      body: jsonEncode({
+        'email': email,
+        'password': password,
+      }),
+    );
+
+    if (response.statusCode == 200) {
+      final responseData = jsonDecode(response.body);
+      final token = responseData['token'];
```
Comment (GoKubo1997) [inline lib/features/sign_in/presentation/sign_in_page.dart]: You want to wrap the operation with the try-catch for unexpected events like unstable networks.

**Review by GoKubo1997 — APPROVED** [review]
Great.

File: lib/features/splash/presentation/splash_page.dart (hunk 2 above)
Comment (nattawutk123) [reply lib/features/splash/presentation/splash_page.dart]: really cool. thank you.

**Pull request merged**